test/extensions/filters/network/common/fuzz/uber_per_writefilter.cc

```diff
@@ -30,6 +32,31 @@ std::vector<absl::string_view> UberWriteFilterFuzzer::filterNames() {
   return filter_names;
 }
 
+void UberWriteFilterFuzzer::checkInvalidInputForFuzzer(const std::string& filter_name,
+                                                       Protobuf::Message* config_message) {
+  // System calls such as reading files are prohibited in this fuzzer. Some inputs that crash the
+  // mock/fake objects are also prohibited. We could also avoid fuzzing some unfinished features by
+  // checking them here. For now there is only one filter {MongoProxy} on which we have a
+  // constraint.
+  const std::string name = Extensions::NetworkFilters::Common::FilterNameUtil::canonicalFilterName(
+      std::string(filter_name));
+  if (filter_name == NetworkFilterNames::get().MongoProxy) {
+    envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy& config =
+        dynamic_cast<envoy::extensions::filters::network::mongo_proxy::v3::MongoProxy&>(
+            *config_message);
+    if (config.has_delay() && config.mutable_delay()->has_header_delay()) {
+      // MongoProxy filter doesn't allow header_delay because it will pass nullptr to percentage()
```

Comment:
The code from .../network/mongo_proxy/proxy.cc#L403 :

```
  // Use a default percentage
  const auto percentage = fault_config_->percentage(nullptr);
```

The code being called in .../filters/common/fault/fault_config.cc :

```
envoy::type::v3::FractionalPercent
HeaderPercentageProvider::percentage(const Http::RequestHeaderMap* request_headers) const {
  const auto header = request_headers->get(header_name_);
  if (header == nullptr) {
    return percentage_;
  }
```

It seems clear that the caller intends to get the default percentage using the null argument, but the code does not handle this correctly.

How about changing `HeaderPercentageProvider::percentage()` to return `percentage_` if `request_headers==nullptr`?  If you do that, then you don't need the check in this file.

Reply:
> /filters/common/fault/fault_config.cc

Actually the caller wants to call this:
```
class FixedDelayProvider : public DelayProvider {
  public:
    FixedDelayProvider(std::chrono::milliseconds delay,
                       const envoy::type::v3::FractionalPercent& percentage)
        : delay_(delay), percentage_(percentage) {}

    // DelayProvider
    absl::optional<std::chrono::milliseconds>
    duration(const Http::RequestHeaderMap*) const override {
      return delay_;
    }

    envoy::type::v3::FractionalPercent percentage(const Http::RequestHeaderMap*) const override {
      return percentage_;
    }
```
And the developer didn't expect it to call on HeaderPercentageProvider(because mongo_proxy will always pass nullptr, which is meaningless for HeaderPercentageProvider), which shares the same base class with FixedDelayProvider.

And I agree that adding a request_header==nullptr would be better.